woocommerce/woocommerce-ios
# [Order Details] Remove payment gateway account and onboarding use from order details

Closes: #7044


### Description

Previously, the Order Details screen used PaymentGatewayAccount to determine whether to show the `Collect [Card] Payment` button. That required it to fetch and refresh the payment gateway accounts, and use them:
1. to show or hide the button, and
2. to present onboarding when required. 

This button now opens a `Payment Method` selection screen, opening up the option for cash payments, and sharing the payment link, alongside Card. That means that Order Details no longer has any need for the payment gateway accounts, nor the card onboarding flow.

The accounts and onboarding dependencies were already unused, so this PR removes both of these dependencies from Order Details.

### Testing instructions


1. Open an unpaid order from the Orders tab.
2. Observe that on Order Details, the `Collect Payment` button is shown
3. Tap it, and observe that the `Payment Method` screen is shown
4. Go back, and check that pull-to-refresh works correctly

### Screenshots

No visible change.

---
